modules/giapi/src/main/scala/giapi/client/gpi/GpiClient.scala

```diff
@@ -98,22 +110,67 @@ final class GpiClient[F[_]](override val giapi: Giapi[F]) extends GiapiClient[F]
       ),
       DefaultCommandTimeout
     )
+
+  override def genericApply(configuration: Configuration): F[CommandResult] = {
+    // TODO Implement a smarter apply
+    def smartApply(): F[CommandResult] =
+      giapi.command(Command(
+                      SequenceCommand.APPLY,
+                      Activity.PRESET_START,
+                      configuration
+                    ),
+                    DefaultCommandTimeout)
+
+    for {
+      _ <- statusDb.value("gpi:fpu") // placeholder
+      a <- smartApply()
+    } yield a
+  }
+}
+
+object GpiClient {
+  // Used for simulations
+  def simulatedGpiClient(ec: ExecutionContext): Resource[IO, GpiClient[IO]] =
+    Resource.liftF(
+      for {
+        c <- Giapi.giapiConnectionIO(ec).connect
+      } yield new GpiClient(c, GiapiStatusDb.simulatedDb[IO])
+    )
+
+  def gpiClient[F[_]: ConcurrentEffect](
+    url:     String,
+    context: ExecutionContext): Resource[F, GpiClient[F]] = {
+    val giapi: Resource[F, Giapi[F]] =
+      Resource.make(
+        Giapi.giapiConnection[F](url, context).connect
+      )(_.close)
+
+    val db: Resource[F, GiapiStatusDb[F]] =
+      Resource.make(
+        GiapiStatusDb.newStatusDb[F](url, List("gpi:heartbeat"))
+      )(_.close)
+
+    for {
+      c <- giapi
+      d <- db
+    } yield new GpiClient[F](c, d)
+  }
```

Comment:
`(giapi, db).mapN(...)` if you prefer. No big deal.

Reply:
👍